### PR TITLE
community/gammu: upgrade to 1.38.5; modernize

### PR DESCRIPTION
**Pull request by TBK**

### CODE REVIEW

File: community/gammu/APKBUILD [hunk 1]
```diff
@@ -14,43 +15,35 @@ makedepends="$depends_dev unixodbc-dev libdbi-dev postgresql-dev
...
-	local i
-	cd "$_builddir"
-	for i in $source; do
-		case $i in
-		*.patch) msg $i; patch -p1 -i "$srcdir"/$i || return 1;;
-		esac
-	done
+	cd "$builddir"
+	default_prepare
 }
```
Comment (rnalrd) [inline community/gammu/APKBUILD]: You can completely omit prepare(), since it's doing only default stuff.

**Comment by TBK** [comment]
Roger 👍 Will fix it straight away.

**Comment by TBK** [comment]
Working on adding check(), currently the test case fails one test - https://github.com/gammu/gammu/issues/303

**Comment by TBK** [comment]
The test is failing because iconv in Alpine does not support `latin1cp437 (latin1 cp437)`.

~3 years ago there was also an issue related to iconv https://bugs.alpinelinux.org/issues/2870 adn this patch was introduced https://git.alpinelinux.org/cgit/aports/commit/?id=fd99da9ad84491f7a9cbdfebe061f76ae43d66e5, not sure what happened to it.

Should the issue be fixed upstream or on our end?

**Comment by TBK** [comment]
Could it be caused by https://git.alpinelinux.org/cgit/aports/tree/main/musl/iconv.c not mentioning the full list of musl supported charsets (https://git.musl-libc.org/cgit/musl/tree/src/locale/codepages.h)?

**Comment by nijel** [comment]
musl apparently does support cp437, but has bad mapping at least for one char, see https://github.com/gammu/gammu/issues/303#issuecomment-345258460

**Comment by TBK** [comment]
@rnalrd The failing test has been disabled, the issue has been reported to musl and this PR is now ready to be merged :)

**Comment by TBK** [comment]
The musl char mapping issues apparently has been resolved 8 days ago in master branch so it will be included in the next musl release - http://www.openwall.com/lists/musl/2017/11/22/2

**Comment by jirutka** [comment]
Merged as 86563770727dd2b567ccb74be1c9737ddec79f68 with few modifications. Thanks!

**Pull request closed without merging**